### PR TITLE
Ziemleszcz/dtr 489

### PR DESCRIPTION
**Pull request by ziemleszcz**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imx6ull).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: adc/ad7779/imx6ull/dma.c [hunk 1]
```diff
@@ -98,16 +82,67 @@ static int sdma_configure(size_t size, size_t count, addr_t *phys_addr)
...
+	return sdma_configure(sdma_common.buffer.size, sdma_common.buffer.count, sdma_common.buffer.phys_addr, sdma_common.buffer.bd_paddr);
+}
+
+
 void dma_free(void)
 {
-	sdma_free_uncached(sdma_common.buffer.ptr, sdma_common.buffer.size * sdma_common.buffer.count);
+	size_t aligned_size = sdma_common.buffer.size;
+	if (aligned_size % _PAGE_SIZE)
+		aligned_size += _PAGE_SIZE - aligned_size % _PAGE_SIZE;
```
Comment (agkaminski) [inline adc/ad7779/imx6ull/dma.c:143]:
```suggestion
	if ((aligned_size % _PAGE_SIZE) != 0) {
		aligned_size += _PAGE_SIZE - aligned_size % _PAGE_SIZE;
	}
```

File: adc/ad7779/imx6ull/dma.c [hunk 2]
```diff
@@ -19,52 +19,36 @@ static struct {
...
-	addr_t bd_paddr;
 	sdma_channel_config_t cfg;
 	sdma_context_t sdma_context;
 
 	uint8_t event_transfer = 39; /* SAI3 RX FIFO */
 	uint8_t event_channel = event_transfer;
 
 	size_t aligned_size = size;
-	if (aligned_size % _PAGE_SIZE)
+	if (aligned_size % _PAGE_SIZE) {
```
Comment (agkaminski) [inline adc/ad7779/imx6ull/dma.c:40]:
```suggestion
	if ((aligned_size % _PAGE_SIZE) != 0) {
```

File: adc/ad7779/imx6ull/dma.c [hunk 3]
```diff
@@ -98,16 +82,67 @@ static int sdma_configure(size_t size, size_t count, addr_t *phys_addr)
...
+
+
+static int sdma_init(size_t size, size_t count, addr_t *phys_addr)
+{
+	int res;
+	unsigned tries = 25;
+	addr_t bd_paddr;
+
+	size_t aligned_size = size;
+	if (aligned_size % _PAGE_SIZE) {
```
Comment (agkaminski) [inline adc/ad7779/imx6ull/dma.c:96]:
```suggestion
	if ((aligned_size % _PAGE_SIZE) != 0) {
```

File: adc/ad7779/imx6ull/dma.c [hunk 4]
```diff
@@ -98,16 +82,67 @@ static int sdma_configure(size_t size, size_t count, addr_t *phys_addr)
...
+		aligned_size += _PAGE_SIZE - aligned_size % _PAGE_SIZE;
+	}
+
+	while ((res = sdma_open(&sdma_common.sdma, SDMA_DEVICE_FILE_NAME)) < 0) {
```
Comment (agkaminski) [inline adc/ad7779/imx6ull/dma.c:100]:
```suggestion
	res = sdma_open(&sdma_common.sdma, SDMA_DEVICE_FILE_NAME);
	while (res < 0) {
```

**Review by agkaminski — COMMENTED** [review]
Style nitpicks, otherwise LGTM

File: adc/ad7779/imx6ull/dma.c (hunk 4 above)
Comment (ziemleszcz) [reply adc/ad7779/imx6ull/dma.c:100]: This one is not good idea.

Comment (gerard5) [reply adc/ad7779/imx6ull/dma.c:100]: I think @agkaminski  meant something similar to this:

```suggestion
	for (;;) {
		res = sdma_open(&sdma_common.sdma, SDMA_DEVICE_FILE_NAME);
		if (res >= 0) {
			break;
		}

		usleep(100 * 1000);
		if (--tries == 0) {
			log_error("failed to open SDMA device file (%s)", SDMA_DEVICE_FILE_NAME);
			return -1;
		}
	}
```

Comment (agkaminski) [reply adc/ad7779/imx6ull/dma.c:100]: `res` is unused, so then
```suggestion
	while (sdma_open(&sdma_common.sdma, SDMA_DEVICE_FILE_NAME) < 0) {
```

**Review by agkaminski — APPROVED** [review]

**Review by nalajcie — COMMENTED** [review]
General approach looks good. 

Added some nitpicks, I believe most of them originate from the intent to keep the changes minimal.

**Review by nalajcie — DISMISSED** [review]
one small remark, otherwise LGTM.

maybe we should also configure DOUT_STRENGTH with value we know works in testing?

**Review by nalajcie — APPROVED** [review]

**Pull request merged**